program-runtime/src/loaded_programs.rs

```diff
@@ -648,14 +648,13 @@ impl<FG: ForkGraph> LoadedPrograms<FG> {
     }
 
     pub fn prune_by_deployment_slot(&mut self, slot: Slot) {
-        self.entries.retain(|_key, second_level| {
+        for second_level in self.entries.values_mut() {
             *second_level = second_level
                 .iter()
                 .filter(|entry| entry.deployment_slot != slot)
                 .cloned()
                 .collect();
```

Comment:
looks like a open-coded vec retain, i.e. could be `second_level.retain(|entry| entry.deployment_slot != slot);`